package.json

```diff
@@ -22,6 +22,7 @@
     "typedoc:publish": "npm run typedoc && gh-pages --dist ./packages/core/api --dest ./api",
     "changeset": "changeset",
     "prerelease": "npm run build && npm run test",
+    "verifyTypescriptExamples": "find packages docs -iname '*.md' -not -path '*/node_modules/*' -not -path '*/.vuepress/*' -not -iname 'CHANGELOG.md' -exec scripts/verifyTypescript \"{}\" + ;",
```

Comment:
Could we encapsulate this on the `bash` side instead and just run that script here? Feels like exposing implementation details in the npm script.

Reply:
The bash script takes a list of files as arguments and verifies those.

Doing it this way means you can easily run the verification for a single file, which I did for fast iteration all the time while developing:

```bash
$ scripts/verifyTypescript docs/guides/typescript.md
```

In vim, what I do is:
```
:nmap ,t :w\|!scripts/verifyTypescript docs/guides/typescript.md<cr>
```

This means, "When I press comma-T in sequence, save the current file and run the typescript verifier."

This makes for a very tight iteration loop.

Reply:
That makes sense and that's some great functionality! Love how easy it is to hook that up in vim too, gonna have to steal that :P 

I guess I'm looking at it from the perspective of:

- Would I understand what this script is supposed to do if I hadn't seen this PR and don't know `find`? The name makes that obvious but the content could be hard to parse.
- Would I know how to use this script if I hadn't seen this PR?

Might just be a detail worth noting in `CONTRIBUTING.md`!

Reply:
I would consider fixing the last 2 issues and just adding this script as a linter to all PRs, but wanted to hear some feedback from @Andarist and @davidkpiano before doing that.

Reply:
Even though this is in bash which I barely know it's still somewhat readable. I wouldn't be able to tweak this beyond some simple copy-pasta-like changes - but I guess there might just not be a need for this any time soon.

TLDR: I'm OK with this for now.